app/celery/v3/notification_tasks.py

```diff
@@ -128,13 +130,23 @@ def v3_process_notification(request_data: dict, service_id: str, api_key_id: str
     return
 
 
-# TODO - retry conditions
-# TODO - error handling
+@notify_celery.task(
+    serializer="pickle",
+    autoretry_for=(AwsSesClientException,),
+    retry_backoff=True,
+    retry_backoff_max=3600,  # 1 hour
+    max_retries=26           # This is about 1 day total of retries.
+)
 @notify_celery.task(serializer="pickle")
 def v3_send_email_notification(notification: Notification, template: Template):
     # TODO - Determine the provider.  For now, assume SES.
-    # TODO - test "client is None"
     client = clients.get_email_client("ses")
+    if client is None:
+        notification.status = NOTIFICATION_TECHNICAL_FAILURE
+        notification.status_reason = "Couldn't get the provider client."
+        db.session.add(notification)
+        db.session.commit()
```

Comment:
Please standardize this into a method, such as the [handle_non_retryable](https://github.com/department-of-veterans-affairs/notification-api/blob/master/app/celery/common.py#L30) method. Per our logging documentation this should be logged as well, another reason for the standard method.

Reply:
You can create a v3 method that does similar, that'd be fine since the `update_notification_status_by_id` used in that method is messy.

Reply:
done